### PR TITLE
Add initial client

### PR DESCRIPTION
**Pull request by daisukixci**
The client is capable of authenticating, manually crafted search and do basic observable search. Note that this is implemented this way until we have the openapi cli generation working in Go

### CODE REVIEW

File: examples/main.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// - search for observables in the Yeti platform.
+// Note that you need to setup the YETI_URL and YETI_API_KEY environment variables to run this example.
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/daisukixci/goyeti"
```
Comment (tomchop) [inline examples/main.go]: Can we update this to point to the `yeti-patform/goyeti` now or do we have to wait for it to be merged first?

File: client.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/daisukixci/goyeti/models"
```
Comment (tomchop) [inline client.go]: Update here too

**Review by tomchop — CHANGES_REQUESTED** [review]

File: examples/main.go (hunk 1 above)
Comment (daisukixci) [reply examples/main.go]: Yes it was a copy of what I did before the repo was created. I fixed it now

File: client.go (hunk 2 above)
Comment (daisukixci) [reply client.go]: fixed

File: client_test.go
Comment (daisukixci) [inline client_test.go:1]: TODO: I need to finish the test

**Review by udgover — APPROVED** [review]
LGTM! Thanks for this contribution!

**Pull request merged**